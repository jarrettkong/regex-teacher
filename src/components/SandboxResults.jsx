import React from 'react'
import './SandboxResults.scss'

const SandboxResults = (props) => {
  return (
    <section className="SandboxResults">
      <div className="SandboxResults-container">
        <h2 className="SandboxResults-header">Matches</h2>
        {props.matches.map((match, i) => {
          return <p key={i} className="SandboxResults-match">{i+1}. {match}</p>
        })}     
      </div>
      <div className="SandboxResults-container">
        <h2 className="SandboxResults-header">Groups</h2>
        <p>Coming Soon</p>
        {/* {props.groups.map((group, i) => {
          return <p className="SandboxResults-match">{i+1}. {group}</p>
        })} */}
      </div>
    </section>
  )
}

export default SandboxResults
