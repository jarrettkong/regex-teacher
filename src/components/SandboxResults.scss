.SandboxResults {
  border: 1px solid white;
  border-radius: 2px;
  padding: 25px;
  display: grid;
  grid-template-rows: repeat(2, 1fr);
}

.SandboxResults-header {
  margin: 0;
}

.SandboxResults-match {
  margin: 5px 20px;
}

.SandboxResults-container {
  overflow-y: scroll;
}