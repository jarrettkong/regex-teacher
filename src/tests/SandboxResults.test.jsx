import React from 'react';
import SandboxResults from '../components/SandboxResults.jsx'
import { shallow } from 'enzyme';

const mockData = ["email1@test.com", "email_two@test-2.com"]

describe('SandboxResults', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SandboxResults matches={mockData}/>
    )
  });

  it('should render the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

});