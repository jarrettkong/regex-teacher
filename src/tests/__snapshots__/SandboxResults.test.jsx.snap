// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SandboxResults should render the snapshot with all the data passed in 1`] = `
<section
  className="SandboxResults"
>
  <div
    className="SandboxResults-container"
  >
    <h2
      className="SandboxResults-header"
    >
      Matches
    </h2>
    <p
      className="SandboxResults-match"
      key="0"
    >
      1
      . 
      email1@test.com
    </p>
    <p
      className="SandboxResults-match"
      key="1"
    >
      2
      . 
      email_two@test-2.com
    </p>
  </div>
  <div
    className="SandboxResults-container"
  >
    <h2
      className="SandboxResults-header"
    >
      Groups
    </h2>
    <p>
      Coming Soon
    </p>
  </div>
</section>
`;
